### PR TITLE
Support fully parallel in Wptrunner

### PR DESCRIPTION
**Pull request by WeizhongX**
One typical usage for devs is to run WPTs from a few local directories. Current grouping method tends to put tests into a few large groups for such scenario, which limits the total number of test runners we could launched, dispite the number of cores the machine had.

The FullyParallelGroupedSource chuck every test into a separate group, thus could utilize as much as possible power the local host had. Test shows running ~500 tests from the same directory now takes ~ 1min, reduced from previous > 2 minutes.

Bug: 41870

### CODE REVIEW

File: tools/wptrunner/wptrunner/wptcommandline.py [hunk 1]
```diff
@@ -592,6 +595,12 @@ def check_args(kwargs):
             print("--test-groups file %s not found" % kwargs["test_groups_file"])
             sys.exit(1)
 
+    if kwargs["fully_parallel"]:
+        if (kwargs["test_groups_file"] is not None or
+            kwargs["run_by_dir"] is not False):
+            print("Can't pass --test-groups or --run-by-dir together with --fully-parallel")
```
Comment (jonathan-j-lee) [inline tools/wptrunner/wptrunner/wptcommandline.py:601]: Maybe a matter of style, but maybe consolidate with the earlier check to make the option constraints obvious (the earlier one doesn't mention `--fully-parallel`)? Something like:

```
if sum([
    kwargs["test_groups_file"] is not None,
    kwargs["run_by_dir"] is not False,
    kwargs["fully_parallel"],
]) > 1:
    print('Must pass up to one of: --test-groups, --run-by-dir, --fully-parallel')
if kwargs["test_groups_file"] is not None and not os.path.exists(...):
    ...
```

**Review by jonathan-j-lee — APPROVED** [review]
LGTM, optional nits

File: tools/wptrunner/wptrunner/wptcommandline.py (hunk 1 above)
Comment (WeizhongX) [reply tools/wptrunner/wptrunner/wptcommandline.py]: Done

File: tools/executive.py
[the remark this answers is not in the document]
Comment (WeizhongX) [reply tools/executive.py:1]: Moved to `manifest/mputil.py`, and renamed to `max_parallelism`

File: tools/wptrunner/wptrunner/wptcommandline.py [hunk 2]
```diff
@@ -61,6 +62,8 @@ def create_parser(product_choices=None):
                         help="Split run into groups by directories. With a parameter,"
                         "limit the depth of splits e.g. --run-by-dir=1 to split by top-level"
                         "directory")
+    parser.add_argument("-f", "--fully-parallel", action='store_true',
+                        help='run all tests in parallel')
```
[the remark this answers is not in the document]
Comment (WeizhongX) [reply tools/wptrunner/wptrunner/wptcommandline.py]: Updated to "Run every test in a separate group for fully parallelism."

**Pull request merged**